### PR TITLE
[yang]: Update AAA yang models

### LINKED ISSUE #9710: [yang-models] sonic-system-aaa.yang missing key in aaa authentication
#### Description
`apply-patch` to config aaa authentication failed due to some key in sonic-system-aaa.yang being unsupported.
Whereas `sudo config aaa authentication` to config has no problem.
```
admin@vlab-01:~$ sudo config aaa authentication -h
Usage: config aaa authentication [OPTIONS] COMMAND [ARGS]...

  User authentication

Options:
  -h, -?, --help  Show this message and exit.

Commands:
  debug        AAA debug [enable | disable | default]
  failthrough  Allow AAA fail-through [enable | disable | default]
  fallback     Allow AAA fallback [enable | disable | default]
  login        Switch login authentication [ {radius, tacacs+, local} |...
  trace        AAA packet trace [enable | disable | default]
admin@vlab-01:~$ sudo config aaa authentication debug enable
admin@vlab-01:~$ sudo config aaa authentication failthrough enable
admin@vlab-01:~$ sudo config aaa authentication fallback enable
admin@vlab-01:~$ sudo config aaa authentication trace enable
admin@vlab-01:~$ show aaa
AAA authentication login local (default)
AAA authentication failthrough True
AAA authentication debug True
AAA authentication fallback True
AAA authentication trace True
AAA authorization login local (default)
AAA accounting login disable (default)

```

#### Steps to reproduce the issue:
1. Add AAA authentication config by `apply-patch`
2. config action fails
3.

#### Describe the results you received:
```
admin@vlab-01:~/tacacs$ sudo config apply-patch -i /FEATURE -i /QUEUE -i /SCHEDULER t1.json
Patch Applier: Patch application starting.
Patch Applier: Patch: [{"op": "add", "path": "/AAA", "value": {"authentication": {"debug": "True", "failthrough": "True", "fallback": "True", "login": "local,tacacs+", "trace": "True"}}}]
...
sonic_yang(3):All Keys are not parsed in AAA
dict_keys(['authentication'])
sonic_yang(3):exceptionList:["'fallback'"]
sonic_yang(3):Data Loading Failed:All Keys are not parsed in AAA
dict_keys(['authentication'])
Failed to apply patch
Usage: config apply-patch [OPTIONS] PATCH_FILE_PATH
Try "config apply-patch -h" for help.

Error: Given patch is not valid because it will result in an invalid config
```
```
dict_keys(['authentication'])
sonic_yang(3):exceptionList:["'trace'"]
sonic_yang(3):Data Loading Failed:All Keys are not parsed in AAA
```
```
libyang[0]: Invalid value "True" in "failthrough" element. (path: /sonic-system-aaa:sonic-system-aaa/AAA/AAA_LIST[type='authentication']/failthrough)
sonic_yang(3):Data Loading Failed:Invalid value "True" in "failthrough" element.
Note: Below table(s) have no YANG models:
BGP_PEER_RANGE, CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, KDUMP, RESTAPI, SNMP, SNMP_COMMUNITY, TELEMETRY,
libyang[0]: Invalid value "True" in "debug" element. (path: /sonic-system-aaa:sonic-system-aaa/AAA/AAA_LIST[type='authentication']/debug)
sonic_yang(3):Data Loading Failed:Invalid value "True" in "debug" element.
```

#### Describe the results you expected:
Expected `apply-patch` action succeed.

#### Output of `show version`:

```
admin@vlab-01:~/tacacs$ show version

SONiC Software Version: SONiC.master.58770-6402a0226
Distribution: Debian 11.1
Kernel: 5.10.0-8-2-amd64
Build commit: 6402a0226
Build date: Mon Dec 13 09:02:14 UTC 2021
Built by: AzDevOps@sonic-build-workers-000Z9S

Platform: x86_64-kvm_x86_64-r0
HwSKU: Force10-S6000
ASIC: vs
ASIC Count: 1
Serial Number: N/A
Model Number: N/A
Hardware Revision: N/A
Uptime: 07:01:22 up 6 days,  3:30,  1 user,  load average: 0.19, 0.30, 0.45

```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by ganglyu**
Signed-off-by: Gang Lv ganglv@microsoft.com

#### Why I did it
AAA yang model is not up to date.

#### How I did it
Add fallback and trace field, and replace boolean_type

#### How to verify it
Run UT for sonic_yang_models.
Follow the steps from #9710 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

#### Description for the changelog

Fix #9710

### CODE REVIEW

**Comment by wen587** [comment]
Verified from local mannually test. It works.
```
admin@vlab-01:~/tacacs$ sudo config apply-patch -i /FEATURE -i /QUEUE -i /SCHEDULER t1.json
Patch Applier: Patch application starting.
Patch Applier: Patch: [{"op": "add", "path": "/AAA", "value": {"authentication": {"debug": "True", "failthrough": "True", "fallback": "True", "login": "local,tacacs+", "trace": "True"}}}]
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
...
Patch Applier: The patch was sorted into 5 changes:
Patch Applier:   * [{"op": "add", "path": "/AAA", "value": {"authentication": {"debug": "True"}}}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/failthrough", "value": "True"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/fallback", "value": "True"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/login", "value": "local,tacacs+"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/trace", "value": "True"}]
Patch Applier: Applying 5 changes in order:
Patch Applier:   * [{"op": "add", "path": "/AAA", "value": {"authentication": {"debug": "True"}}}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/failthrough", "value": "True"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/fallback", "value": "True"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/login", "value": "local,tacacs+"}]
Patch Applier:   * [{"op": "add", "path": "/AAA/authentication/trace", "value": "True"}]
Patch Applier: Verifying patch updates are reflected on ConfigDB.
Patch Applier: Patch application completed.
Patch applied successfully.

```

**Review by qiluo-msft — APPROVED** [review]
Please wait some days for more eyes.

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang [hunk 1]
```diff
@@ -32,17 +36,28 @@ module sonic-system-aaa {
...
                 leaf debug {
-                    type boolean;
+                    type stypes:boolean_type;
                     description "Enable/disable AAA debugging";
-                    default false;
+                    default False;
                 }
 
+                leaf trace {
+                    type stypes:boolean_type;
```
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-aaa.yang:57]: Is there any control plane component acting on this info ?
How could we prevent this from getting persisted in config_db.json file ?

Comment (renukamanavalan) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:57]: From Gang Lv:
cli commands and config db both have this 'trace' fie

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**